source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp

```diff
@@ -620,21 +621,58 @@ namespace RendererQml
+
+		auto textBackgroundTag = std::make_shared<QmlTag>("Rectangle");
+		textBackgroundTag->Property("color", "'transparent'");
+		contentItemTag->Property("background", textBackgroundTag->ToString());
+		contentItemTag->Property("onEditingFinished", Formatter() << "{ if(Number(text) < " << inputId << ".from || Number(text) > " << inputId << ".to){\nremove(0,length)\nif(" << inputId << ".hasDefaultValue)\ninsert(0, " << inputId << ".defaultValue)\nelse\ninsert(0, " << inputId << ".from)\n}\n}");
         contentItemTag->Property("color", context->GetColor(AdaptiveCards::ForegroundColor::Default, false, false));
 
+		auto colorOverlayTag = std::make_shared<QmlTag>("ColorOverlay");
+		colorOverlayTag->Property("anchors.fill", "parent");
+		colorOverlayTag->Property("source", "parent");
+		colorOverlayTag->Property("color", context->GetColor(AdaptiveCards::ForegroundColor::Default, false, false));
+
+		auto upIndicatorTag = std::make_shared<QmlTag>("Rectangle");
+		upIndicatorTag->Property("width", "20");
+		upIndicatorTag->Property("height", "parent.height/2");
+		upIndicatorTag->Property("x", inputId + ".mirrored ? 0 : parent.width - width");
+		upIndicatorTag->Property("radius", "5");
+		upIndicatorTag->Property("color", inputId + ".up.pressed ? '#08000000' : 'transparent'");
+
+		auto upIndicatorImage = std::make_shared<QmlTag>("Image");
+		upIndicatorImage->Property("source", Formatter() << "\"" << RendererQml::arrow_up_12 << "\"");
+		upIndicatorImage->Property("anchors.centerIn", "parent");
+		upIndicatorImage->AddChild(colorOverlayTag);
+		upIndicatorTag->AddChild(upIndicatorImage);
+
+		auto downIndicatorTag = std::make_shared<QmlTag>("Rectangle");
+		downIndicatorTag->Property("width", "20");
+		downIndicatorTag->Property("height", "parent.height/2");
+		downIndicatorTag->Property("x", inputId + ".mirrored ? 0 : parent.width - width");
+		downIndicatorTag->Property("y", "parent.height/2");
+		downIndicatorTag->Property("radius", "5");
+		downIndicatorTag->Property("color", inputId + ".down.pressed ? '#08000000' : 'transparent'");
+
+		auto downIndicatorImage = std::make_shared<QmlTag>("Image");
+		downIndicatorImage->Property("source", Formatter() << "\"" << RendererQml::arrow_down_12 << "\"");
+		downIndicatorImage->Property("anchors.centerIn", "parent");
+		downIndicatorImage->AddChild(colorOverlayTag);
+		downIndicatorTag->AddChild(downIndicatorImage);
+
 		auto doubleValidatorTag = std::make_shared<QmlTag>("DoubleValidator");
 
 		auto uiNumberInput = std::make_shared<QmlTag>("SpinBox");
 		uiNumberInput->Property("id", input->GetId());
 		uiNumberInput->Property("width", "parent.width");
+		uiNumberInput->Property("padding", "0");
 		uiNumberInput->Property("stepSize", "1");
 		uiNumberInput->Property("editable", "true");
 		uiNumberInput->Property("validator", doubleValidatorTag->ToString());
 
 		if (input->GetMin() == std::nullopt)
 		{
-			input->SetMin(INT_MIN);
+			input->SetMin(0);
```

Comment:
Why are we setting `SetMin` to `0` hardcoded? because this will then not allow to enter -ve numbers.